perfdash/parser.go

```diff
@@ -245,23 +245,36 @@ type latencyMetric struct {
 }
 
 type schedulingMetrics struct {
-	PredicateEvaluationLatency  latencyMetric `json:"predicateEvaluationLatency"`
-	PriorityEvaluationLatency   latencyMetric `json:"priorityEvaluationLatency"`
-	PreemptionEvaluationLatency latencyMetric `json:"preemptionEvaluationLatency"`
-	BindingLatency              latencyMetric `json:"bindingLatency"`
-	E2eSchedulingLatency        latencyMetric `json:"e2eSchedulingLatency"`
-	ThroughputAverage           float64       `json:"throughputAverage"`
-	ThroughputPerc50            float64       `json:"throughputPerc50"`
-	ThroughputPerc90            float64       `json:"throughputPerc90"`
-	ThroughputPerc99            float64       `json:"throughputPerc99"`
+	PredicateEvaluationLatency  histogramVec `json:"predicateEvaluationLatency"`
+	PriorityEvaluationLatency   histogramVec `json:"priorityEvaluationLatency"`
+	PreemptionEvaluationLatency histogramVec `json:"preemptionEvaluationLatency"`
+	BindingLatency              histogramVec `json:"bindingLatency"`
+	SchedulingAlgorithmLatency  histogramVec `json:"schedulingAlgorithmLatency"`
+	E2eSchedulingLatency        histogramVec `json:"e2eSchedulingLatency"`
 }
 
-func parseOperationLatency(latency latencyMetric, operationName string) perftype.DataItem {
-	perfData := perftype.DataItem{Unit: "ms", Labels: map[string]string{"Operation": operationName}, Data: make(map[string]float64)}
-	perfData.Data["Perc50"] = float64(latency.Perc50) / float64(time.Millisecond)
-	perfData.Data["Perc90"] = float64(latency.Perc90) / float64(time.Millisecond)
-	perfData.Data["Perc99"] = float64(latency.Perc99) / float64(time.Millisecond)
-	return perfData
+func parseOperationLatency(histogramVecMetric histogramVec, operationName string) perftype.DataItem {
```

Comment:
IIUC, this changes are backward compatible, your're just adding a new metric (SchedulingAlgorithmLatency) and refactoring this method to make it more generic.

If that's the case, could you lease send the perf-dash changes in a separate PR? We'll submit and deploy a new version of it and verify that the change doesn't break anything (unfortunately, we don't have any e2e tests for perf-dash). Once it's done we can submit the CL2 changes

Reply:
+1

Can we refactor the existing `parseHistogramMetric` function to make it more generic?

https://github.com/kubernetes/perf-tests/blob/982668b933540721dfc713627697fbdc6e9af5c2/perfdash/parser.go#L326